### PR TITLE
Add a db test

### PR DESCRIPTION
**Pull request by MarinMihaiAlexandru**
Added a basic cucumber test that creates a table, inserts values into it and then asserts that the correct values are added. 
It uses an H2 in-memory db

This shoould resolve issue #15

### CODE REVIEW

File: src/test/java/com/java/project/cucumber/DBSteps.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            // Handle errors for Class.forName
+            e.printStackTrace();
+        } finally {
+            // finally block used to close resources
+            try {
+                if (stmt != null) stmt.close();
+            } catch (SQLException se2) {
+            } // nothing we can do
+            try {
+                if (conn != null) conn.close();
```
Comment (VoicuGeo) [inline src/test/java/com/java/project/cucumber/DBSteps.java:134]: conn.close() will be easier to read it if will be on the next line. My opinion. Try also with "Ctrl+Alt+l "

Comment (MarinMihaiAlexandru) [reply src/test/java/com/java/project/cucumber/DBSteps.java:130]: I like to have them both on 1 line if the clause is 1 line and doesn't become too long. We can discuss this further with the team and if it's agreed then I will change it

Comment (CiprianBucur) [reply src/test/java/com/java/project/cucumber/DBSteps.java:130]: I also prefer different lines also having "{" and "}".

File: src/test/java/com/java/project/cucumber/DBSteps.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            // STEP 4: Extract data from result set
+            while (rs.next()) {
+                assertEquals("Wrong ID", Integer.valueOf(id), rs.getInt("id"));
+                assertEquals("Wrong Age", Integer.valueOf(age), rs.getInt("age"));
+                assertEquals("Wrong First Name", firstName, rs.getString("first_name"));
+                assertEquals("Wrong Last Name", lastName, rs.getString("last_name"));
+            }
+            // STEP 5: Clean-up environment
+            rs.close();
+        } catch (SQLException se) {
```
Comment (CiprianBucur) [inline src/test/java/com/java/project/cucumber/DBSteps.java:117]: Can "catch" be on the next line?

File: src/test/java/com/java/project/cucumber/DBSteps.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            // Handle errors for JDBC
+            se.printStackTrace();
+        } catch (Exception e) {
```
Comment (CiprianBucur) [inline src/test/java/com/java/project/cucumber/DBSteps.java:120]: Can "catch" be on the next line?

File: src/test/java/com/java/project/cucumber/DBSteps.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+            // Handle errors for Class.forName
+            e.printStackTrace();
+        } finally {
+            // finally block used to close resources
+            try {
+                if (stmt != null) stmt.close();
+            } catch (SQLException se2) {
```
Comment (CiprianBucur) [inline src/test/java/com/java/project/cucumber/DBSteps.java:127]: Can "catch" be on the next line?

File: src/test/java/com/java/project/cucumber/DBSteps.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+            } // nothing we can do
+            try {
+                if (conn != null) conn.close();
+            } catch (SQLException se) {
```
Comment (CiprianBucur) [inline src/test/java/com/java/project/cucumber/DBSteps.java:131]: Can "catch" be on the next line?

File: src/test/java/com/java/project/cucumber/DBSteps.java [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            }
+            // STEP 5: Clean-up environment
+            rs.close();
+        } catch (SQLException se) {
+            // Handle errors for JDBC
+            se.printStackTrace();
+        } catch (Exception e) {
+            // Handle errors for Class.forName
+            e.printStackTrace();
+        } finally {
```
Comment (CiprianBucur) [inline src/test/java/com/java/project/cucumber/DBSteps.java:123]: Can "finally" be on the next line?

File: pom.xml [hunk 7]
```diff
@@ -57,6 +57,16 @@
             <version>2.2</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+            <version>2.2.7.RELEASE</version>
```
Comment (CiprianBucur) [inline pom.xml:63]: there is a new version "2.3.0.RELEASE". Can "2.3.0.RELEASE" be used?

File: src/test/java/com/java/project/cucumber/DBSteps.java [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+            // finally block used to close resources
+            try {
+                if (stmt != null) stmt.close();
+            } catch (SQLException se2) {
+            } // nothing we can do
+            try {
+                if (conn != null) conn.close();
```
Comment (CiprianBucur) [inline src/test/java/com/java/project/cucumber/DBSteps.java:82]: I would prefer:
if (conn != null) { 
    conn.close();
}

File: src/test/java/com/java/project/cucumber/DBSteps.java [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+            } catch (SQLException se) {
```
Comment (CiprianBucur) [inline src/test/java/com/java/project/cucumber/DBSteps.java:83]: Can "catch" be on the next line?

File: src/test/java/com/java/project/cucumber/DBSteps.java [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+        } catch (SQLException se) {
+            // Handle errors for JDBC
+            se.printStackTrace();
+        } catch (Exception e) {
+            // Handle errors for Class.forName
+            e.printStackTrace();
+        } finally {
+            // finally block used to close resources
+            try {
+                if (stmt != null) stmt.close();
```
Comment (CiprianBucur) [inline src/test/java/com/java/project/cucumber/DBSteps.java:126]: Same as above
{
    stmt.close();
}

**Pull request merged**